compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9066,11 +9065,12 @@ private STNode parseIntermediateClause(boolean isRhsExpr) {
             case ASCENDING_KEYWORD:
             case DESCENDING_KEYWORD:
                 return parseOrderByClause(isRhsExpr);
+            case LIMIT_KEYWORD:
+                return parseLimitClause(isRhsExpr);
```

Comment:
shall we update the grammar rule in the doc comment above this method with limit clause?

Reply:
fixed in : f2064f7